### PR TITLE
Remove deprecated API - HttpSender and RemoteBaggMana

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to https://github.com/jaegertracing/jaeger-client-java/pull/414#discussion_r190146876

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java [hunk 1]
```diff
@@ -41,33 +41,6 @@
   private final Restriction invalidRestriction;
   private final Restriction validRestriction;
 
-  /**
-   * @deprecated use {@link Builder}
-   */
-  @Deprecated
-  public RemoteBaggageRestrictionManager(
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java]: That's not good, then.

File: jaeger-core/src/main/java/io/jaegertracing/senders/HttpSender.java [hunk 2]
```diff
@@ -38,19 +38,6 @@
...
-  /**
-   * @param endpoint Jaeger REST endpoint consuming jaeger.thrift, e.g
-   * http://localhost:14268/api/traces
-   *
-   * Uses the default {@link okhttp3.OkHttpClient} which uses {@link okhttp3.ConnectionPool#ConnectionPool()}.
-   * Use {@link HttpSender.Builder} if you need to add more parameters
-   * @deprecated use {@link Builder}
-   */
-  @Deprecated
-  public HttpSender(String endpoint) {
```
[the remark this answers is not in the document]
Comment (jpkrohling) [reply jaeger-core/src/main/java/io/jaegertracing/senders/HttpSender.java]: Not good :)

**Comment by pavolloffay** [comment]
I am not sure if people are overriding it. We can make builder constructors provided is needed.

**Comment by pavolloffay** [comment]
>  If the HttpSender is part of the public API, then it should be extensible.

I am not quite sure about ^^. There are public APIs which are not extensible. Maybe our question is: is there a reason to extend it? Do people need it?

 From the API design perspective, visibility should be restricted as much as possible. If people ask for it make it public

**Comment by jpkrohling** [comment]
> There are public APIs which are not extensible

True, but then they marked as "final". 

> Maybe our question is: is there a reason to extend it? Do people need it?

That's indeed the question. There's only one method that could be overridden, so, one could just extend `ThriftSender` directly, just like the `HttpSender` does.

File: jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (black-adder) [reply jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java]: oops, my bad, we shouldn't deprecate this

[the remark this answers is not in the document]
Comment (pavolloffay) [reply jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java]: We want to deprecate it and use builder instead. Does protected constructor work for you?

[the remark this answers is not in the document]
Comment (black-adder) [reply jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java]: that's fine

[the remark this answers is not in the document]
Comment (pavolloffay) [reply jaeger-core/src/main/java/io/jaegertracing/baggage/RemoteBaggageRestrictionManager.java]: thanks the private with the most parameters is changed from private to `protected`
```java
protected RemoteBaggageRestrictionManager(
       String serviceName,
       BaggageRestrictionManagerProxy proxy,
       Metrics metrics,
       boolean denyBaggageOnInitializationFailure,
       int refreshIntervalMs,
       int initialDelayMs
   ) {
```

**Pull request merged**